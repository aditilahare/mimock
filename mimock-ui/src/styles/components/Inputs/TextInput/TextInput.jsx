import React from 'react';
import PropTypes from 'prop-types';
import { InputBlock, FormInput, ErrorMessage } from './TextInput.style';

function TextInput({
	name,
	type,
	dataTestId,
	placeHolder,
	error,
	errorMessage,
	value,
	onChange,
}) {
	return (
		<InputBlock>
			<FormInput
				type={type}
				data-testid={dataTestId}
				placeholder={placeHolder}
				$error={error}
				value={value}
				onChange={onChange}
			/>
			<If condition={error}>
				<ErrorMessage data-testid={`input-error-${name}`}>
					{errorMessage}
				</ErrorMessage>
			</If>
		</InputBlock>
	);
}

TextInput.propTypes = {
	name: PropTypes.string.isRequired,
	type: PropTypes.string,
	dataTestId: PropTypes.string.isRequired,
	placeHolder: PropTypes.string.isRequired,
	error: PropTypes.bool,
	errorMessage: PropTypes.string,
	value: PropTypes.string,
	onChange: PropTypes.func,
};

TextInput.defaultProps = {
	type: 'text',
	error: false,
	errorMessage: '',
	value: '',
	onChange: () => {},
};

export default TextInput;
