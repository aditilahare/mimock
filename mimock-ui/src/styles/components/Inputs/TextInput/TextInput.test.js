import React from 'react';
import { act, render } from '@testing-library/react';
import TextInput from './TextInput';

describe('TextInput', () => {
	it('should render text input', async () => {
		let tree;
		await act(async () => {
			tree = await render(
				<TextInput
					name='text'
					dataTestId='text-input'
					placeHolder='Enter text'
				/>
			);
		});

		const { getByTestId, container } = tree;

		expect(getByTestId('text-input')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('should accept input with value', async () => {
		let tree;
		await act(async () => {
			tree = await render(
				<TextInput
					name='text'
					dataTestId='text-input'
					placeHolder='Enter text'
					value='test'
				/>
			);
		});

		const { getByTestId, container } = tree;

		expect(getByTestId('text-input')).toHaveValue('test');

		expect(container).toMatchSnapshot();
	});
});
