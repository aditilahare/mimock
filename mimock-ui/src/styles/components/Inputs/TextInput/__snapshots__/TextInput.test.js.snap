// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should accept input with value 1`] = `
<div>
  <div
    class="block h-12 min-h-12 max-h-12"
  >
    <input
      class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
      data-testid="text-input"
      placeholder="Enter text"
      type="text"
      value="test"
    />
  </div>
</div>
`;

exports[`TextInput should render text input 1`] = `
<div>
  <div
    class="block h-12 min-h-12 max-h-12"
  >
    <input
      class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
      data-testid="text-input"
      placeholder="Enter text"
      type="text"
      value=""
    />
  </div>
</div>
`;
